packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFNavigationDelegateHostApi.m

```diff
@@ -179,6 +220,40 @@ - (void)webViewWebContentProcessDidTerminate:(WKWebView *)webView {
                                              NSAssert(!error, @"%@", error);
                                            }];
 }
+
+- (void)webView:(WKWebView *)webView
+    didReceiveAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge
+                    completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition,
+                                                NSURLCredential *_Nullable))completionHandler {
+  [self.navigationDelegateAPI
+      didReceiveAuthenticationChallengeForDelegate:self
+                                           webView:webView
+                                         challenge:challenge
+                                        completion:^(FWFAuthenticationChallengeResponse *response,
+                                                     FlutterError *error) {
+                                          NSAssert(!error, @"%@", error);
```

Comment:
Why is this an assert? Couldn't a plugin client have an implementation on the Dart side that throws an exception, leading to a `FlutterError` here?

Reply:
This is a pattern I started with all the FlutterApi methods. IIRC when there is an exception thrown on the Dart side during a FlutterApi method, this error is returned and it wouldn't be printed anywhere. Essentially failing silently. This would catch the error and print it to the console when the app is not in release mode or disabling asserts.

At the very least, this used to be the case on Android.

Reply:
Right, but in most cases I thought it was just calling our own code, so we could reasonably assert that an error didn't exist.

I guess the question here is: if a plugin *client* throws an exception in their own Dart code that implements the handler, do we want the entire application to crash?

Reply:
@bparrishMines Should we adjust this, or land as-is and revisit later if necessary?

Reply:
I created https://github.com/flutter/flutter/issues/140410 to track the discussion. I think we could probably find a better alternative, but I haven't seen any issues raised by users.